ui/src/table.tsx

```diff
@@ -705,18 +705,41 @@ export const
       [groupByKey, setGroupByKey] = React.useState('*'),
       contentRef = React.useRef<Fluent.IScrollablePane | null>(null),
       tableRef = React.useRef<{ resetSortIcons: () => void } | null>(null),
+      sortRef = React.useRef<{ column: WaveColumn, sortAsc: B } | null>(null),
```

Comment:
There's no particular reason, only to prevent additional re-render when setting the state.

Reply:
> prevent additional re-render when setting the state.

Where should the extra re-render occur? Since React batches state updates, this shouldn't happen (both `sort` and `filter` functions set the state already).

Reply:
The problem with the previous implementation was the stale closure of `filter` function which was not getting the up-to-date sort state and therefore could not perform correct sorting after filtering.